bokehjs/src/coffee/renderer/tile/mercator_tile_source.coffee

```diff
@@ -4,30 +4,25 @@ TileSource = require "./tile_source"
 class MercatorTileSource extends TileSource
 
   type: 'MercatorTileSource'
-  
-  defaults: =>
-    return _.extend {}, super(), {
-      tile_size : 256
-      max_zoom : 30
-      min_zoom : 0
-      extra_url_vars : {}
-      full_extent : [-20037508.34, -20037508.34, 20037508.34, 20037508.34]
-      x_origin_offset : 20037508.34
-      y_origin_offset : 20037508.34
-      initial_resolution : undefined
-      resolutions : undefined
-    }
-
-  constructor: ->
-    super
-
-    # compensation for not knowing how to do computed default properties
-    # warning: order matters here...
+
+  initialize: (options) ->
+
+    super(options)
+
     if not @get('initial_resolution')?
-      @set('initial_resolution', 2 * Math.PI * 6378137 / @get('tile_size'))
+      @set('initial_resolution', 2 * Math.PI * 6378137 / @get('tile_size'), {silent: true})
```

Comment:
the better way to solve the "computed defaults" problem I think is to have a separate computed_initial_resolution in the view (or as a computed property) and then make it depend on the developer-set attributes (which are left alone). If you do it as above (overwriting what the developer provides), it doesn't update dynamically later. for example if tile_size changes and initial_resolution was never set, then we should recompute, right? But since you've overwritten initial_resolution you no longer know it was never set.